pkg/kube/controllers/boshdeployment/reconciler.go

```diff
@@ -97,6 +116,16 @@ func (r *ReconcileBOSHDeployment) Reconcile(request reconcile.Request) (reconcil
 		return reconcile.Result{}, err
 	}
 
+	// TODO Need to update instanceState after finishing Variable Generation stuff
+
+	instanceState = "Variable Interpolation"
```

Comment:
Each state needs to happen asynchronously. Reconcile happens, we look at state - make some action.
When that action is complete, reconcile has to happen again, and we need to update state. Given that we are in a different state, we take a different action. 
Variable interpolation should happen in a job, since you don't have variables generated yet.